package/MDAnalysis/lib/util.py

```diff
@@ -2552,3 +2556,59 @@ def wrapper(self, *args, **kwargs):
                         self._kwargs[key] = arg
         return func(self, *args, **kwargs)
     return wrapper
+
+
+def optional_import(
```

Comment:
How about a separate `lib.optional` module? It's a well-characterized problem so I'd prefer than throwing even more into util. We can then also have better/more focused docs to go along with it.